google/cloud/bigquery/magics/magics.py

```diff
@@ -596,6 +596,25 @@ def _cell_magic(line, query):
             _handle_error(error, args.destination_var)
             return
 
+        # Check if query is given as a reference to a variable.
+        if query.startswith("$"):
+            query_var_name = query[1:]
+
+            if query_var_name.isidentifier():
+                ip = IPython.get_ipython()
+                try:
+                    query = ip.user_ns[query_var_name]
+                except KeyError:
+                    raise NameError(
+                        f"Unknown query, variable {query_var_name} does not exist."
+                    ) from None
```

Comment:
Should we chain this error, or is that likely to be more confusing than helpful for `IPython` users?

Reply:
My thinking was that `KeyError` is just an implementation detail here, as it could easily be written in a LBYL style, too (although that would require one extra dict lookup).

Reply:
You could avoid the extra lookup via:

```python
                query = ip.user_ns.get(query_var_name)
                if query is None:
                    raise NameError(
                        f"Unknown query, variable {query_var_name} does not exist."
                    )
```

Reply:
Right... should probably not work such late hours. :upside_down_face: 

(side note - we would still need to distinct between the case when the variable is missing and when the user explicitly set it to `None`, but a sentinel default value would solve that easily)

Anyhow, let me know if there's a strong preference for LBYL over EAFP here and I'll refactor if needed.